docs/docs/secrets-managers.mdx

```diff
@@ -0,0 +1,142 @@
+VAULT_TOKEN # token to authenticate to the vault server
+VAULT_RASA_SECRETS_PATH # path to the secrets in the vault server, if not set it defaults to `rasa-secrets`
+VAULT_TRANSIT_MOUNT_POINT # if transit secrets engine is enabled, set this to  mount point of the transit engine
+```
+
+The `endpoints.yml` configuration for vault secrets manager looks like this:
+```yaml-rasa {3} title="endpoints.yml
+secrets_manager:
+    type: vault
+    token: <token used to authenticate to Vault>
+    url: "http://localhost:1234"
+    secrets_path: rasa-secrets  # if not set it defaults to `rasa-secrets`
+    transit_mount_point: transit # if you have enabled transit secrets engine, and you want to use it
+```
+
+#### How to combine environment variables and `endpoints.yml`
+A simple example on how to combine environment variables and `endpoints.yml` configuration file
+would be to store access token in the environment variable and the rest of the configuration
+in the `endpoints.yml` file.
+
+```bash
+# environment variables
+VAULT_TOKEN=<token used to authenticate to Vault>
+```
+
+
+```yaml-rasa {3} title="endpoints.yml
+secrets_manager:
+    type: vault
+    url: "http://localhost:1234"
+    secrets_path: rasa-secrets  # if not set it defaults to `rasa-secrets`
+    transit_mount_point: transit # if you have enabled transit secrets engine, and you want to use it
+```
+
+### Example
+This is a simple example which demonstrates how to store credentials for tracker store in Vault
+and how to access them from Rasa.
+
+1. Start Vault instance
+    ```bash
+    docker run --cap-add=IPC_LOCK --name=vault -p 1234:1234 -p 8200:8200 -e 'VAULT_DEV_ROOT_TOKEN_ID=myroot' -e 'VAULT_DEV_LISTEN_ADDRESS=0.0.0.0:1234' -d vault
+    ```
+
+2. Store credentials for tracker store in Vault
+    ```bash
+    # go inside Vault container
+    docker exec -it vault sh
+
+    # (optional) if you get access denied error
+    export VAULT_ADDR=http://127.0.0.1:1234
+    export VAULT_TOKEN=myroot
+
+    # store secret in the Vault under the group rasa-secrets
+    vault kv put -mount=secret rasa-secrets sql_store_username=postgres sql_store_password=12345
+    ```
+
+3. Start Postgres container
+    ```bash
+    docker run --name postgres -e POSTGRES_DB=rasa -e POSTGRES_PASSWORD=12345 -p 5432:5432 -d postgres
+    ```
+
+4. Configure Rasa to access the Vault instance
+
+    Checkout the [How to configure access to Vault](#how-to-configure-access-to-vault) section for more details.
+
+5. Configure Rasa to use the Vault secrets manager to fetch credentials for the tracker store
+    ```yaml-rasa {3} title="endpoints.yml
+        tracker_store:
+              type: SQL
+              url: localhost:5432
+              username:
+                    source: secrets_manager.vault
+                    secret_key: sql_store_username
+              password:
+                    source: secrets_manager.vault
+                    secret_key: sql_store_password
+    ```
+
+6. Start Rasa
```

Comment:
What do we mean here by `Start Rasa`? Does it mean Rasa server needs to be shut down when adding secrets to the HashiCorp Vault? Will the changes in endpoint.yml get applied if the user restarts rasa pod? I'm thinking here about our customers that would want to implement HashiCorp Vault into their current deployment

Reply:
We should be more specific here, this example is essential to set up the Vault and use it properly.

Reply:
Rasa does not need to be stopped when secrets to Vault are added.
Rasa only needs to be stopped when we are adding/updating Vault connection configuration as Rasa currently does not support hot-reloading of it's configuration.

Reply:
@Balowen 
I have updated the example to be more comprehensive.
Let me know what do you think.

Reply:
I think it's better now but I also agree with @l2nguyen [here](https://github.com/RasaHQ/rasa/pull/12110#discussion_r1132194604). It's unlikely our customers will ever run it on local.